src/assets/wise5/classroomMonitor/dataExport/strategies/AbstractComponentDataExportStrategy.ts

```diff
@@ -0,0 +1,309 @@
+    workgroupId: number,
+    userId1: number,
+    userId2: number,
+    userId3: number,
+    studentName1: string,
+    studentName2: string,
+    studentName3: string,
+    periodName: string,
+    componentRevisionCounter: ComponentRevisionCounter,
+    componentState: any
+  ): string[];
+
+  generateComponentHeaderRow(columnNames: string[]): string[] {
+    return [...columnNames];
+  }
+
+  populateColumnNames(): void {
+    const defaultColumnNames = [
+      '#',
+      'Workgroup ID',
+      'User ID 1',
+      'Student Name 1',
+      'User ID 2',
+      'Student Name 2',
+      'User ID 3',
+      'Student Name 3',
+      'Class Period',
+      'Project ID',
+      'Project Name',
+      'Run ID',
+      'Start Date',
+      'End Date',
+      'Server Timestamp',
+      'Client Timestamp',
+      'Node ID',
+      'Component ID',
+      'Component Part Number',
+      'Step Title',
+      'Component Type',
+      'Component Prompt',
+      'Student Data',
+      'Component Revision Counter',
+      'Is Submit',
+      'Submit Count'
+    ];
+    for (let c = 0; c < defaultColumnNames.length; c++) {
+      const defaultColumnName = defaultColumnNames[c];
+      this.columnNameToNumber.set(defaultColumnName, c);
+      this.columnNames.push(defaultColumnName);
+    }
+  }
+
+  generateComponentWorkRows(component: Component): string[] {
+    let rows = [];
+    let rowCounter = 1;
+    for (const workgroupId of this.configService.getClassmateWorkgroupIds()) {
+      const rowsForWorkgroup = this.generateComponentWorkRowsForWorkgroup(
+        workgroupId,
+        component.nodeId,
+        component.id,
+        rowCounter
+      );
+      rows = rows.concat(rowsForWorkgroup);
+      rowCounter += rowsForWorkgroup.length;
+    }
+    return rows;
+  }
+
+  private generateComponentWorkRowsForWorkgroup(
+    workgroupId: number,
+    nodeId: string,
+    componentId: string,
+    rowCounter: number
+  ): string[] {
+    const rows = [];
+    const userInfo = this.configService.getUserInfoByWorkgroupId(workgroupId);
+    const userIdsAndStudentNames = new UserIdsAndStudentNames(
+      userInfo.users,
+      this.canViewStudentNames
+    );
+    const componentRevisionCounter = new ComponentRevisionCounter();
+    const componentStates = this.teacherDataService.getComponentStatesByWorkgroupIdAndComponentId(
+      workgroupId,
+      componentId
+    );
+    for (let c = 0; c < componentStates.length; c++) {
+      const componentState = componentStates[c];
+      if (this.shouldExportRow(componentState, c, componentStates.length)) {
+        const row = this.generateComponentWorkRow(
+          rowCounter,
+          workgroupId,
+          userIdsAndStudentNames.getUserId(1),
+          userIdsAndStudentNames.getUserId(2),
+          userIdsAndStudentNames.getUserId(3),
+          userIdsAndStudentNames.getStudentName(1),
+          userIdsAndStudentNames.getStudentName(2),
+          userIdsAndStudentNames.getStudentName(3),
```

Comment:
Pass in ```UserIdsAndStudentNames``` object instead of individual userId1,userId2,...? Do this all the way down the callee until it's actually needed in setStudentIDsAndNames(), i.e. ```generateComponentWorkRow() -> createStudentWorkExportRow() -> setStudentIDsAndNames()```?
```suggestion
    new UserIdsAndStudentNames(
      userInfo.users,
      this.canViewStudentNames
    ),
```